examples/simpack_xrd_dyn_AlGaAs.py

```diff
@@ -0,0 +1,178 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Copyright (c) 2016-2023 Dominik Kriegner <dominik.kriegner@gmail.com>
+
+
+
+# This is an example of the modelling and fitting of the diffraction data obtained on epitaxial heterostructure
+# The design is as follow
+#               _________________________
+#
+#                       GaAs 320 nm          using the same Ga cell as GaAs below (same growth rate)
+#               _________________________
+#
+#                  Al_0.80GaAs 1.8 micron    using the same Ga & Al cells as Al_0.80GaAs below (same growth rate and composition)
+#               _________________________
+#
+#                       GaAs 200 nm           
+#               _________________________
+#
+#                  Al_0.80GaAs 2 micron
+#               _________________________
+#
+#              GaAs substrate and buffer (001)
+#
+# Xray set-up: X'PERT Panalytical with
+#       copper anode,focusing mirror, Ge(220)x4 monochromator, Ge(220)x2 triple axis detection
+# Copyright (c) 2023 C2N - CNRS Author:	  Aristide Lemaître <aristide.lemaitre@c2n.upsaclay.fr>
+
+
+
+import xrayutilities as xu
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def x_Al(aper,apar,mat):
+    # determination of Al composition from c parameter, poisson coefficient and Veegard's law
+    c11 = mat.cij[0,0] # approx of C11 and C12 from alloy with 
+    c12 = mat.cij[0,1]
+    eps_star = (apar - aper)/apar
+    eps_AlAsGaAs = (AlAs_.a - GaAs_.a)/GaAs_.a
+    x = -c11 * eps_star / eps_AlAsGaAs / (c11 * (1 + eps_star) + 2 * c12) 
+    return x
+
+
+# read experimental file
+
+file ='data/simpack_xrd_AlGaAs.xrdml'
+scan = xu.io.panalytical_xml.XRDMLFile(file)
+om = scan.scans[0]['Omega']
+Int = scan.scans[0].int
+
+
+# Omega offset
+offset = -0.047634587753920085
+om_off = om + offset
+
+
+# definition of materials
+
+GaAs_ = xu.materials.GaAs
+AlAs_ = xu.materials.AlAs
+AlGaAs80 = xu.materials.AlGaAs(0.20)
+
+
+# definition of heterostructure layers
+
+sub = xu.simpack.Layer(GaAs_, float('inf'))
+AlGaAs80_1 = xu.simpack.Layer(AlGaAs80, 20000, relaxation=0.)
+GaAs_2 = xu.simpack.Layer(GaAs_, 2000, relaxation=0.)
+AlGaAs80_3 = xu.simpack.Layer(AlGaAs80, 18000, relaxation=0.)
+GaAs_4 = xu.simpack.Layer(GaAs_, 3200, relaxation=0.0)
+
+
+# heterostructure content
+heterostructure_ = AlGaAs80_1 \
+                  +GaAs_2 \
+                  +AlGaAs80_3\
+                  +GaAs_4
+
+# turned into a pseudomorphic structure grown along the 001 direction
+heterostructure = xu.simpack.PseudomorphicStack001('Hetereostructure', sub, heterostructure_)
+
+# X-Ray energy
+wavelength = xu.wavelength('CuKa1')
+en = xu.energy('CuKa1')
```

Comment:
this is the default wavelength and it should therefore be ok not to set it below. for the wavelength I think it is ok since I see currently also no better way to get Cmono otherwise

Reply:
done